internal/fwserver/server_importresourcestate.go

```diff
@@ -53,6 +53,13 @@ func (s *Server) ImportResourceState(ctx context.Context, req *ImportResourceSta
 		return
 	}
 
+	//TODO: add logging for replacing deferred reason
+	if s.deferred != nil {
+		resp.Deferred = &resource.Deferred{
+			Reason: resp.Deferred.Reason,
+		}
+	}
```

Comment:
See comment in `ReadResource`, same idea applies.

SDKv2 logic: hhttps://github.com/hashicorp/terraform-plugin-sdk/blob/e90e8a2497cb1bced1d5614e65c904453244d18c/helper/schema/grpc_provider.go#L1223-L1263

Your logic for constructing an unknown value will be also be different here since it's not using `cty`.. It also will not need to do the error checking for type name. But the goal is to return a single unknown object state for `ImportedResources`